nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,159 @@
+                    // having one. This should not happen.
+                    let message = format!(
+                        "target blueprint with id {} was removed. There is no \
+                        longer any target blueprint",
+                        old.id
+                    );
+                    error!(&log, "{}", message);
+                    json!({"error": message})
+                }
+                (None, Ok(Some((_, new_target)))) => {
+                    // We've found a new target blueprint for the first time.
+                    // Save it and notify any watchers.
+                    let target_id = new_target.id.to_string();
+                    let time_created = new_target.time_created.to_string();
+                    info!(
+                        log,
+                        "found new target blueprint (first find)";
+                        "target_id" => &target_id,
+                        "time_created" => &time_created
+                    );
+                    self.last = Some(Arc::new(new_target));
+                    self.tx.send_replace(self.last.clone());
+                    json!({
+                        "target_id": target_id, "time_created": time_created
+                    })
+                }
+                (Some(old), Ok(Some((_, new)))) => {
+                    let target_id = new.id.to_string();
+                    let time_created = new.time_created.to_string();
+                    if old.id != new.id {
+                        // The current target blueprint has been updated
+                        info!(
+                            log,
+                            "found new target blueprint";
+                            "target_id" => &target_id,
+                            "time_created" => &time_created
+                        );
+                        self.last = Some(Arc::new(new));
+                        self.tx.send_replace(self.last.clone());
+                        json!({
+                            "target_id": target_id,
+                            "time_created": time_created
+                        })
+                    } else {
+                        // The new target id matches the old target id
+                        //
+                        // Let's see if the blueprints hold the same contents.
+                        // It should not be possible for the contents of a
+                        // blueprint to change, but we check to catch possible
+                        // bugs further up the stack.
+                        if **old != new {
+                            let message = format!(
+                                "blueprint for id {} changed. \
+                                Blueprints are supposed to be immutable.",
+                                target_id
+                            );
+                            error!(&log, "{}", message);
+                            json!({"error": message})
```

Comment:
Similar to above: if we're going to keep the existing one in this case, then I feel like we should include that information in the JSON object too.

Reply:
Added in [5d72471](https://github.com/oxidecomputer/omicron/pull/4891/commits/5d72471ad30c517f19c94d1eab9274fde74bf61d)